### PR TITLE
Ch6

### PR DESCRIPTION
**Pull request by tonyslowdown**

### CODE REVIEW

File: exercises-solutions/src/main/scala/Ch6.scala [hunk 1]
```diff
@@ -1,13 +1,182 @@
...
+
+    object Problem2 {
+
+      implicit val functorIndexedSeq = new Functor[IndexedSeq] {
+        override def map[A, B](fa: IndexedSeq[A])(f: A ⇒ B): IndexedSeq[B] = fa.map(f)
+      }
+
+      implicit val evenFilterableIndexedSeq = new FilterableWithFilter[IndexedSeq] {
+        def evenFilter[A](p: A => Boolean)(fa: IndexedSeq[A]): IndexedSeq[A] = {
+          if (fa.filter(t => !p(t)).length % 2 == 1) IndexedSeq() // Can we do this?
```
Comment (winitzki) [inline exercises-solutions/src/main/scala/Ch6.scala]: Not sure this is the right implementation... need to think. I expected the implementation to be defined through the filter function on IndexedSeq like this,

```scala
evenFilter(p)(fa) = fa.filter(t => ...) 
```

File: chapter07/src/test/scala/example/Chapter07_01_examplesSpec.scala [hunk 2]
```diff
@@ -4,15 +4,18 @@ import cats.kernel.Semigroup
 import cats.{Functor, Monoid, derive}
```
Comment (winitzki) [inline chapter07/src/test/scala/example/Chapter07_01_examplesSpec.scala]: Get rid of this and Flattenable, it's not from this chapter ;)

**Review by winitzki — APPROVED** [review]

File: chapter07/src/test/scala/example/Chapter07_01_examplesSpec.scala (hunk 2 above)
Comment (tonyslowdown) [reply chapter07/src/test/scala/example/Chapter07_01_examplesSpec.scala]: i didnt write that though lol. that's just brought in from merging from upstream, which is your repo

Comment (tonyslowdown) [reply chapter07/src/test/scala/example/Chapter07_01_examplesSpec.scala]: Ah maybe you removed it in yours, and since I didn't rebase, it's like this.

**Review by winitzki — APPROVED** [review]

File: exercises-solutions/src/main/scala/Ch6.scala [hunk 3]
```diff
@@ -1,13 +1,519 @@
...
+      type C[A] = Option[A] => Int
+
+      implicit val contrafunctorC = new Contravariant[C] {
+        override def contramap[A, B](ca: C[A])(f: B ⇒ A): C[B] = {
+          case Some(b: B) => ca(Some(f(b)))
+          case None => ca(None)
+        }
+      }
+
+      implicit val contrafilterableC = new ContraFilterable[C] {
```
Comment (winitzki) [inline exercises-solutions/src/main/scala/Ch6.scala]: It was not necessary to assume that C is contra-filterable.

File: exercises-solutions/src/main/scala/Ch6.scala [hunk 4]
```diff
@@ -1,13 +1,519 @@
...
+        override def inflate[A](ca: C[A]): C[Option[A]] = {
+          case Some(optA: Option[A]) => ca(optA)
+          case None => ca(None)
+        }
+      }
+
+      def deflateC[A](cOptA: C[Option[A]]): C[A] = { (optA: Option[A]) =>
```
Comment (winitzki) [inline exercises-solutions/src/main/scala/Ch6.scala]: Instead, we need to show that `deflate` can be defined for a contrafunctor.

So, we want a general definition of a PTVF with this type signature:

```scala
def deflate[C[_]: Contravariant, A]: C[Option[A]] => C[A] = { ... }
```

which should work for any `Contravariant` `C`.

**Review by winitzki — APPROVED** [review]

**Pull request merged**